lib/feedPage.dart

```diff
@@ -0,0 +1,713 @@
+                                    ),
+                                    SizedBox(
+                                      height: 1.0,
+                                    ),
+                                    Text(
+                                      'Feed',
+                                      style: TextStyle(
+                                          fontSize: 12, color: Colors.white),
+                                    ),
+                                  ],
+                                ),
+                              ),
+                            ),
+                            Tab(
+                              child: Padding(
+                                padding: const EdgeInsets.fromLTRB(
+                                    17.0, 7.0, 17.0, 3.0),
+                                child: Column(
+                                  children: [
+                                    Icon(
+                                      Icons.bookmark,
+                                      color: Colors.white,
+                                      size: 18.0,
+                                    ),
+                                    SizedBox(
+                                      height: 1.0,
+                                    ),
+                                    Text(
+                                      'Bookmarks',
+                                      style: TextStyle(
+                                          fontSize: 12, color: Colors.white),
+                                    ),
+                                  ],
+                                ),
+                              ),
+                            )
+                          ],
+                        ),
+                      ),
+                    ),
+                  ],
+                ),
+              ],
+            ),
+          )
+        ],
+      ),
+    );
+  }
+}
+
+//////////////////////////////////////////////////////////////////////////////
+
+// code for the upper design of appbar
+
+class CurveClipper extends CustomClipper<Path> {
+  @override
+  Path getClip(Size size) {
+    Path path = Path()
+      // set the "current point"
+      ..addArc(Rect.fromLTWH(0, 0, size.width / 2, size.width / 3), pi, -1.57)
+      ..lineTo(9 * size.width / 10, size.width / 3)
+      ..addArc(
+          Rect.fromLTWH(
+              size.width / 2, size.width / 3, size.width / 2, size.width / 3),
+          pi + 1.57,
+          1.57)
+      ..lineTo(size.width, 0)
+      ..lineTo(0, 0)
+      ..lineTo(0, size.width / 6);
+    return path;
+  }
+
+  @override
+  bool shouldReclip(oldCliper) => false;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+
+// code for the sidebar
+
+class NavDrawer extends StatefulWidget {
+  @override
+  _NavDrawerState createState() => _NavDrawerState();
+}
+
+class _NavDrawerState extends State<NavDrawer> {
+  bool isSwitched1 = true;
```

Comment:
These variables can be put in a list and accessed using their indices to ensure easy and handy adding of more categories in future.